### ISSUE TITLE
IndexError on empty list in Compose

### ISSUE THREAD

**Issue by montanier** [albumentations-team/albumentations#1462]
## 🐛 Bug

When the "masks" field of a sample is an empty list, the composition is running in a `IndexError: list index out of range` error. 

This error can be traced back to the following commit: https://github.com/albumentations-team/albumentations/commit/a51a6410c8c3635db93c7c6ddf3244e5a44bc738

## To Reproduce

Run the following code:
```python
import numpy as np
from albumentations.core.composition import Compose
from albumentations.augmentations import Crop

img_shape = (100, 100)

image = np.ones(img_shape)
masks = []

transform = Compose([Crop(always_apply=False, p=1.0, x_min=20, y_min=50, x_max=30, y_max=60)])
res = transform(image=image, masks=masks)
print(res)
```

## Expected behavior

I would expect to be able to run the above example without error. Like for version `1.3.0` I think I should obtain:
```python
{'image': array([[1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]]), 'masks': []}
```

## Environment

 Albumentations version 1.3.1

## Additional thoughts

From the original commit it is hard to understand what was exactly the bug, and how the dev fixes it. What about documenting the original bug and adding some unit test?

Maybe somewhere it should be documented what is expected when a field is "empty": an empty list, an empty tensor or a None?

**Comment by montanier** [comment]
An addition to this bug, the commit actually introduced some inconsistencies between the kwargs of `transform`. If one introduce bboxes with the modification bellow:

```python
import numpy as np
from albumentations.core.composition import Compose
from albumentations.augmentations import Crop

img_shape = (100, 100)

image = np.ones(img_shape)
masks = None
bboxes = None

transform = Compose([Crop(always_apply=False, p=1.0, x_min=20, y_min=50, x_max=30, y_max=60)],
                    bbox_params={'format': 'pascal_voc', min_area': 0.0, 'min_visibility': 0.1, 'min_width': 0.0, 'min_height': 0.0, 'check_each_transform': True})
res = transform(image=image, masks=masks, bboxes=bboxes)
print(res)
```

One get the following error:
```
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    res = transform(image=image, masks=masks, bboxes=bboxes)
  File "/home/moj2rng/Code/aps_training/.venv/lib/python3.8/site-packages/albumentations/core/composition.py", line 199, in __call__
    p.ensure_data_valid(data)
  File "/home/moj2rng/Code/aps_training/.venv/lib/python3.8/site-packages/albumentations/core/bbox_utils.py", line 112, in ensure_data_valid
    data_exists = data_name in data and len(data[data_name])
TypeError: object of type 'NoneType' has no len()
```

Therefore with albumentations 1.3.1 the way to work with unlabelled data would be:
```python
import numpy as np
from albumentations.core.composition import Compose
from albumentations.augmentations import Crop

img_shape = (100, 100)

image = np.ones(img_shape)
masks = None
bboxes = []

transform = Compose([Crop(always_apply=False, p=1.0, x_min=20, y_min=50, x_max=30, y_max=60)],
                    bbox_params={'format': 'pascal_voc', 'min_area': 0.0, 'min_visibility': 0.1, 'min_width': 0.0, 'min_height': 0.0, 'check_each_transform': True})
res = transform(image=image, masks=masks, bboxes=bboxes)
print(res)
```

I think one way should be chosen, applied everywhere, and tested.

**Comment by Dipet** [comment]
Fixed  with #1494
